activestorage/lib/active_storage/analyzer/image_analyzer.rb

```diff
@@ -34,7 +34,13 @@ def metadata
       def read_image
         download_blob_to_tempfile do |file|
           require "mini_magick"
-          yield MiniMagick::Image.new(file.path)
+          image = MiniMagick::Image.new(file.path)
+          if image.valid?
+            yield image
+          else
+            logger.info "Skipping image analysis because mini_magick doesn't support the file"
```

Comment:
I'm tempted to move the `rescue LoadError` here since having the logging happen in the two methods makes it a bit harder to read, if we like this approach I'm happy to amend the PR.

Reply:
Please do.

This message is misleading: it’s ImageMagick that fails to decode the image.

Reply:
Please do move the `LoadError` rescue down.